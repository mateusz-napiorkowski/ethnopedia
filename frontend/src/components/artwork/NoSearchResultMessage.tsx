import { AiOutlineFileSearch } from "react-icons/ai";

const NoSearchResultMessage = () => {
    return (
        <div className="px-4 max-w-screen-xl pt-10 pb-10 py-4 bg-white dark:bg-gray-800 shadow-md w-full rounded-lg mb-4 border border-gray-300 dark:border-gray-600 text-center">
            <AiOutlineFileSearch className="mx-auto w-16 h-16 mb-4 text-gray-400" />
            <p className="text-xl mb-4">Brak wyników.</p>
            <p className="text-md text-gray-600 dark:text-gray-300">
                Nie znaleziono rekordów pasujących do Twojego wyszukiwania. Spróbuj zmienić reguły wyszukiwania.
            </p>
        </div>
    );
};

export default NoSearchResultMessage;
