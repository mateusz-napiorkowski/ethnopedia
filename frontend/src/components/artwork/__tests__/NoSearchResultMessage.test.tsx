import '@testing-library/jest-dom';
import { render } from '@testing-library/react'
import NoSearchResultMessage from "../NoSearchResultMessage"
import { MemoryRouter, Routes, Route } from "react-router-dom";

const renderComponent = () => {
    return render(
        <MemoryRouter initialEntries={[`/`]}>
            <Routes>
                <Route path="/" element={<NoSearchResultMessage/>}/>
            </Routes>  
        </MemoryRouter>
    );
};

describe("NoSearchResultMessage tests", () => {
    it("should render NoSearchResultMessage component", () => {
        const {container} = renderComponent()
        expect(container).toMatchSnapshot()
    })
})