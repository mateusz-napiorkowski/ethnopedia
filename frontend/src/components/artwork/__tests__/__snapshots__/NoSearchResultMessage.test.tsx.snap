// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSearchResultMessage tests should render NoSearchResultMessage component 1`] = `
<div>
  <div
    class="px-4 max-w-screen-xl pt-10 pb-10 py-4 bg-white dark:bg-gray-800 shadow-md w-full rounded-lg mb-4 border border-gray-300 dark:border-gray-600 text-center"
  >
    <svg
      class="mx-auto w-16 h-16 mb-4 text-gray-400"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M688 312v-48c0-4.4-3.6-8-8-8H296c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8zm-392 88c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H296zm144 452H208V148h560v344c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V108c0-17.7-14.3-32-32-32H168c-17.7 0-32 14.3-32 32v784c0 17.7 14.3 32 32 32h272c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm445.7 51.5l-93.3-93.3C814.7 780.7 828 743.9 828 704c0-97.2-78.8-176-176-176s-176 78.8-176 176 78.8 176 176 176c35.8 0 69-10.7 96.8-29l94.7 94.7c1.6 1.6 3.6 2.3 5.6 2.3s4.1-.8 5.6-2.3l31-31a7.9 7.9 0 0 0 0-11.2zM652 816c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
      />
    </svg>
    <p
      class="text-xl mb-4"
    >
      Brak wyników.
    </p>
    <p
      class="text-md text-gray-600 dark:text-gray-300"
    >
      Nie znaleziono rekordów pasujących do Twojego wyszukiwania. Spróbuj zmienić reguły wyszukiwania.
    </p>
  </div>
</div>
`;
